src/unix/ftrace.c

```diff
@@ -1807,48 +1832,59 @@ ftrace_init(unix_heaps uh, filesystem fs)
 
     /* nop tracer */
     current_tracer = &(tracer_list[0]);
-
+   for (int i = 0; i < MAX_CPUS; i++) {
+        cpuinfo ci = cpuinfo_from_id(i);
+        if (ftrace_cpu_init(ci) != 0)
+            return -1;
+    }
     return 0;
 }
 
 void
 ftrace_deinit(void)
 {
+    for (int i = 0; i < MAX_CPUS; i++) {
+        cpuinfo ci = cpuinfo_from_id(i);
+        ftrace_cpu_deinit(ci);
+    }
     deallocate_http_listener(ftrace_heap, ftrace_hl);
 }
 
 int
-ftrace_thread_init(thread t)
+ftrace_cpu_init(cpuinfo ci)
```

Comment:
This isn't related to this PR, but it would be nice if we could standardize per-cpu inits so that they can just be called altogether as CPUs come online. There's a timer-specific cpu init called in new_cpu(), but it could just become a list. (filed https://github.com/nanovms/nanos/issues/1328)